htlcswitch/hop/payload.go

```diff
@@ -129,6 +129,104 @@ type Payload struct {
 	metadata []byte
 }
 
+// PackHopPayload writes to the passed io.Writer, the series of byes that can
+// be placed directly into the per-hop payload (EOB) for this hop. This will
+// include the required routing fields, as well as serializing any of the
+// passed optional TLVRecords.  nextChanID is the unique channel ID that
+// references the _outgoing_ channel ID that follows this hop. This field
+// follows the same semantics as the NextAddress field in the onion: it should
+// be set to zero to indicate the terminal hop.
+func PackHopPayload(w io.Writer, p Payload) error {
```

Comment:
This method pretty much mirrors a function in the `route` package and is used only a test utility right now. Think about it a bit more.